### ISSUE TITLE
Change translational dither frequency

### ISSUE THREAD

**Issue by humnaawan** [LSSTDESC/ObsStrat#15]
Based on conversations at the Project & Community Workshop 2018, it was agreed that we'd like to _not_ dither in between visits to the same field in the same night for better calibrations. This calls for less frequent translational dithers than per visit, as are currently implemented as descDithers. Rotational dithers frequency (after every filter change) is okay for now.

In an attempt to address this, the translational dithers need to be changed to perNight dithers. #3, #5

**Comment by rbiswas4** [comment]
> Based on conversations at the Project & Community Workshop 2018, it was agreed that we'd like to not dither in between visits to the same field in the same night for better calibrations

I heard about the requirement that they should not be dithered between pairs, but if there are more visits to the same location in WFD, I was under the impression that they could be dithered. Maybe I missed something else, which would ensure that they not be dithered during the night. Irrespective, single dithers through the night might just be an efficient implementation that takes care of the visit pair arguments, as well as sufficient dithers for large scale structure.

> Rotational dithers frequency (after every filter change) is okay for now.
@humnaawan : Maybe I don't follow what you are saying here. If you say you will dither once the filters are changed, that is certainly sufficient to have un-dithered visit pairs for most of the cadences where the visit pairs are in the same filter. However, if we include observing strategies where the visits pairs are in different filters (as I hope we will), dithering at the end of a filter change will mean two visits in a pair are dithered, which we don't want..

**Comment by humnaawan** [comment]
@rbiswas4 yes, you're right. We dont _have_ to dither only once per night but very rarely does a field get visited more than twice in a given night. Also, per night translational dithers would bring an added benefit of being more comparable to altSched which changes its HEALPix gridding every night.

For LSS systematics, more frequent dithers are the best but there's merit to not dithering in between pairs and there isn't much gain in varying dithers during the night (we looked at two variant of nightly dithers in [our](https://arxiv.org/abs/1605.00555) paper: perNight and FieldPerNight which lead to comparable effectiveness).

**Comment by rbiswas4** [comment]
@humnaawan Great ... so one translational dither per night at the same point is an efficient implementation keeping everyone happy. 

In terms of rotational dithers, I thought we had agreed that things would not change within a pair of visits. But if they change after a filter change this could happen. That is the other thing I was worried about.

**Comment by humnaawan** [comment]
Right, we'd ideally want to not do rotational dithers in the between pairs either but that'd require a bit of work, especially since there has been only one implementation of rotational dithers.

Since rotational dithers are still largely under investigation, I am unsure of whether the overhead of implementing a "clever" rotational dither strategy is justified right now, especially since even for the one implementation that we have right now,  we do not have any quantitative study of the effectiveness of the dithers. I am interested in exploring this but have a few things that have priority.

@egawiser thoughts? I am not sure who else I should be tagging.

**Comment by egawiser** [comment]
I think the issue of rotational dithering is more complicated, because asking visit pairs to be observed at the same *orientation* is actually a request for a rotational dither, although often modest, to be performed to make that happen.  But I agree that this will be what is desired for visit pairs, and since we're pushing for visit pairs to occur in different filters, making large rotational dithers at each filter change no longer sounds like a great approach.  For a quick improvement, we could try adopting a single random rotational position (RotTelPos) at the beginning of each night and *reverting* to it after each filter change that night and investigate if that produces sufficiently uniform distributions of the two rotation angles for Weak Lensing systematics to be reduced acceptably.  But when there's time to put some real thought into this, it would be great to try to code up a requirement that the 2nd observation of a field in a filter that night be performed at the same RotSkyPos value as the 1st one (that's what we want to cover the same regions of the field on the same chips, and that will automatically create some variation in RotTelPos, so it's only RotSkyPos whose distribution should become less uniform this way).
